Feature/Add remaining chart type slide generators

Nadimchouaib opened the pull request:
This PR introduces new slide generation functions to support additional chart types in the presentation pipeline:

- `generate_tile_slide()`: Creates column-style tile charts to summarize activity duration per group.
- `generate_circle_slide()`: Shows per-category values using scaled circle markers (e.g., for reunion class satisfaction).
- `generate_horizontal_bar_slide():` Displays average hours and subjective values across multiple activities in a horizontal bar format.
Each function follows the existing pattern used in the pipeline.

Commit:
chore: update input data with new categories and cleaned values

Commit:
feat: Add generate_circle_slide() function for circle based chart slides

Creates a reusable function to generate slides with circle charts, visualizing a single metric across categories (e.g., reunion class). Supports optional units, subset filtering, custom ordering.

Commit:
chore: Rename variable in generate_circle_slide

Commit:
feat: Add tile chart slide generator with focal and comparison modes

Supports dynamic circle sizing, label padding, and margin scaling based on number of activities and character width.

Commit:
Chore: Update README file with tile charts documentation

Commit:
chore: Update tile function generator documentation

Commit:
chore: add mapping file

Add mapping file to easily map displayed label names to pipeline column names.

Commit:
chore: update simulated_pipeline_input.csv

update pipeline data input file with more activities hours and subjective values

Commit:
feat: add generate_horizontal_bar_slide function

Introduce horizontal bar chart function using variable_label_map.csv to map internal variable names to display labels. Supports dual-axis charts for hours and subjective values.

Commit:
chore: update README file with horizontal bar slide generation documentation

Commit:
chore: update instructions list formatting

Commit:
chore: remove extra space

ahmedjoubest commented on R/tile_function.R:
We won't process the column variables to create the label. We'll use the mapping file (column name --> label) as agreed on. This is the case for:
- Tile chart
- Bar metric chart
- Any other chart where the column names is pre-processed.

**We want to avoid this kind of text processing.**

ahmedjoubest commented on README.md:
Earlier in the documentation, it's worth noting the logic with the mapping file.

ahmedjoubest commented on R/instructions.R:
```diff
@@ -312,7 +525,418 @@ instructions <- list(
+  
+  list(
+    function_name = "generate_horizontal_bar_slide",
+    metric = c(
+      "Volunteering",
+      "Therapy",
+      "Games",
+      "School/learning",
+      "Side Projects",
+      "Job searching",
+      "Napping",
+      "Hobbies"
+    ),
```
By convention in programming, metrics are typically referenced by their IDs rather than their labels.

ahmedjoubest commented on R/instructions.R:
```diff
@@ -312,7 +525,418 @@ instructions <- list(
+  # ------ Slide 34: Activities hours vs subjective values for focal group -----
+  list(
+    function_name = "generate_horizontal_bar_slide",
+    metric = c(
+      "Volunteering",
+      "Therapy",
+      "Games",
+      "School/learning",
+      "Side Projects",
+      "Job searching",
+      "Napping",
+      "Hobbies"
+    ),
```
Same. By convention in programming, metrics are typically referenced by their IDs rather than their labels.

ahmedjoubest replied:
@Nadimchouaib 
We identified a problem during the call: `subjective_value = TRUE` should not be allowed. The instruction is not explicit enough.

If there’s no subjective value provided, `subjective_value` should be `NULL`. Otherwise, the user must explicitly provide the vector of subjective columns. We agreed that there should be no implicit logic or text-based guessing — everything must be explicit.

Example (correct usage):

```r
list(
  function_name = "generate_horizontal_bar_slide",
  metric = c("Volunteering", "Therapy", "Games", "School/learning", "Side Projects", "Job searching", "Napping", "Hobbies"),
  subjective_value = c(subjective_value_1, 2, 3, ...),  # must be defined explicitly
  title = "DISCRETIONARY TIME – ACTIVITIES",
  x_title = c("Avg. Hours Per Week", "Avg. Subjective Value"),
  y_title = "Activity",
  focal_group = list(name = focal_group, subset = NULL)
)
```

ahmedjoubest reviewed (changes requested):
Almost! Thank you!

Commit:
fix: enforce metric ID usage and explicit subjective_value

Switch to ID based metric handling and enforced explicit subjective_value (must be vector or NULL).

Commit:
fix: remove text processing in tile chart and support dynamic units

- Use mapping file for all labels in tile chart.
- Display unit from instruction list.

Commit:
feat: update instructions list with unit for tile function

Commit:
fix: remove label text processing in bar chart,

Use mapping file for all metric labels in bar chart and avoid all implicit text transformations.

Commit:
feat: Add `jam_type_distribution` to pipeline data

Extend the data preprocessing pipeline to include a new `jam_type_distribution` column. This supports the donut slide
visualization based on categorical distribution.

Commit:
feat: Add donut chart slide generator with corresponding instructions

Create a new `generate_donut_slide()` function to visualize category
distribution as a ring-style pie chart. Supports:

- Focal and comparison groups
- Placeholder groups for layout consistency
- Vertical white separators between donuts

Commit:
chore: Update README file with donut function documentation

Commit:
chore : clarify role of mapping_file.csv

Commit:
chore: Update CSVs with revised metrics and their corresponding labels.

Commit:
feat: Add generate_line_slide function for multi-metric line charts

Implements a slide generator that creates line charts across categories, supports multiple metrics and optional units.

Commit:
chore: Update REAME file with generate_line_slide() documentation

Commit:
chore: update pipline CSV with more metrics and categories

Commit:
feat: Add generate_bar_stacked_slide() to render stacked distribution slides

Implements generate_bar_stacked_slide() for visualizing category proportions across x-axis groups with optional metric annotations
above each bar. Supports category ordering and subset filtering.

Commit:
feat: Enable multi-category_y comparison in stacked bar charts

Supports side-by-side stacking for multiple category_y variables.

Commit:
feat: Extend generate_bar_stacked_slide() to support category_x subsetting and dynamic metric labels

- Support multiple category_x definitions with subset filtering (e.g., "Kids", "No Kids") with side-by-side bar rendering per x group using subset conditions
- Display top labels for each subset bar in the top based on subset$value
- Improve average metric logic:
  - Per bar when category_x includes subsetting
  - Per x group otherwise
- Validate mutually exclusive use of multi category_x and multi category_y

Commit:
fix : Update the structure of category_x

Commit:
feat: Add horizontal stacked bar chart generator and update slide instructions

Introduce `generate_bar_stacked_horizontal_slide()` to render grouped horizontal stacked bar charts with optional subset and avg support.

Updated slide instruction lists to include use cases based on gender, child status, and work hours.

Commit:
docs: attach latest generated PPTX from pipeline

Commit:
chore: Update README file with stacked bar functions documentation

Nadimchouaib commented:
Hello @ahmedjoubest, 

Thanks for your feedback, I've addressed all the remaining comments mentioned above and also added new slide generation functions to support additional chart types: 

- `generate_donut_slide()`: Builds donut-style pie charts for category distributions with support for comparisons.
- `generate_line_slide()`: Creates line charts for one or more metrics across a categorical x-axis.
- `generate_stacked_vertical_slide()`: Produces vertical stacked bar charts, optionally grouped and annotated with averages.
- `generate_stacked_horizontal_slide()`: Produces horizontal stacked bar charts with subgrouping and optional average labels.

Each function follows the existing pattern used in the pipeline.

Commit:
feat: Add generate_activity_scatter_slide() for scatter plot slide generation

- Introduced new function to generate labeled scatter plot slides from paired metric inputs.
- Added example usage to the instructions list.
- Documented function behavior, features, and usage in the README.

Commit:
chore: Initialize renv for project dependency management

Commit:
fix: apply early filtering for category_x values in stacked bar slide

Displays only the selected bars defined via category_x$value

Commit:
chore : Update README file

Update generate_horizontal_bar_slide provided examples

Commit:
fix: Unify font styling across all slides

Ensures consistent appearance by applying default font (Arial Bold) to all text elements.

Commit:
feat: add support for switching density chart y-axis between % and fraction

Implemented logic to dynamically format the y-axis as either percentages or fractions based on the specified unit.

Commit:
fix: correct unequal bar heights caused by rounding labels instead of data

This update rounds the actual data values (not just the labels) to ensure consistent
bar heights when values are equal.

Commit:
fix: show x-axis line in horizontal bar chart facets

Commit:
fix: Improve bar spacing and rotate x-axis labels to align with y-axis title

- Reduced spacing between intra-group bars (e.g., Men vs Women) for tighter grouping.
- Increased spacing between group sets for clearer visual separation.
- Rotated axis group labels 180°.

Commit:
fix: singular hour notation in duration labels

Use “1 hr” instead of “1 hrs” when referencing a single hour to maintain grammatical accuracy. If `unit` is NULL, the unit is omitted entirely.

Commit:
fix: standardize numeric formatting for uniform and readable numbers.

Implemented consistent formatting rules across all chart functions that print summary values:
- Used sprintf("%.1f") to display values with one decimal (e.g., 6 → "6.0")
- Used sprintf("%.0f") to round and display whole numbers when needed (e.g., 6.8 → "7")

Commit:
fix: round y-axis percent labels to whole numbers in density charts

Commit:
feat: Apply ordinal suffix styling to reunion_class

Ensure reunion_class column is styled with <sup> suffixes.

Commit:
chore: remove extra space

Commit:
fix: Stretch chart to better utilize slide space

Commit:
chore: update list of instructions

Commit:
feat: Add safeguard to handle missing values (NA) in key variables

- Add logic to convert Inf and "" values to NAs
- Implemented NA filtering to exclude rows with missing or non-finite values in key categorical and metric variables used for plotting.
- Ensures cleaner visualization and prevents unexpected chart behavior.

Commit:
chore: Add latest generated slide deck from pipeline output

Commit:
feat: Add safeguard to handle missing columns in required inputs

Commit:
fix: Update missing columns logic to display all instruction messing column at the same time

Commit:
improve documentation

Commit:
correct data path

Commit:
documentation

Commit:
chore: add dataset containing NA and Inf values for testing

Commit:
fix: improve vertical bar stretching logic for better scaling

Replaced the previous bar height scaling approach with a more adaptive method to ensure consistent proportions.

Commit:
feat: Implement safeguard fallback for missing labels

Addinglogic to use the variable name when the mapping file has no label.

Optimizing the safeguard functionality using Sari's data and instructions to ensure proper handling of NAs without breaking slide generation.

Commit:
feat: Add logic to print warnings immediately during slide generation

Commit:
chore: Clean and document main pipeline and instructions

Commit:
chore: Clean and refactor

Commit:
docs: Update README to clarify mapping file fallback behavior

Commit:
fix: Adjust density comparison label display

Commit:
feat: prevent y-axis jumps in `generate_bar_metric_slide` with new placeholder logic

Reworked placeholder handling to be driven directly from the instructions list,
ensuring y-axis limits are consistent across all groups and slide sequences.

Commit:
fix: adapt normalize_na_tbl for read_csv data frames

Update NORMALIZE NAs logic to handle tibble class returned by `read_cs`v, ensuring consistent NA normalization across numeric,
character, and factor columns.

Commit:
feat: update target logic to prevent y-axis jump

Ensure that target lines are also considered when computing y_max, avoiding visual jumps when flipping slides.

Commit:
docs: update RAEDME file

Commit:
fix: correct tile slide logic to compute mean values first and show ordered hours

- Compute mean activity values per group (focal or comparison)
- Select top/bottom n activities based on value metrics
- Display associated hours, ordered for clarity
- Use instructions list directly for mapping instead of external file

Commit:
feat: update instructions list to support new tile slide logic

Commit:
feat: update input data to support new tile slide logic

Commit:
chore: last generated output
